### PR TITLE
[18.0][MIG] mail_notification_with_history: Migration to 18.0

### PR DESCRIPTION
**Pull request by trisdoan**
### Note
- template `message_notification_email` was moved to `data/mail_templates.xml` in https://github.com/odoo/odoo/commit/78c68f4b885c204a1453f80670d46d0ecc53d02c

- Later splitted for better isolate notification layouts in https://github.com/odoo/odoo/commit/a480f1f3cd770a60fca35901d547ed51e56300b7

### This change
- Since t-raw was flagged as deprecated, replace it with t-out
 
### Result
mail_notification_migration

### CODE REVIEW

File: mail_notification_with_history/data/mail_data.xml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+      <t
+                t-set="message_history"
+                t-value="message._get_notification_message_history()"
+            />
+      <t t-if="message_history">
+        <p class="message_history_title">Discussion history in Odoo:</p>
+        <ul>
+          <t t-foreach="message_history" t-as="msg">
+            <li class="message_history_item">
+            <p><t t-esc="msg.author_id.name or 'unknown'" /> commented on <span
```
Comment (henrybackman) [inline mail_notification_with_history/data/mail_data.xml]: `t-esc` is soft deprecated, could we switch to `t-out` here?

**Review by henrybackman — APPROVED** [review]
LG with one minor comment

File: mail_notification_with_history/data/mail_data.xml (hunk 1 above)
Comment (simahawk) [reply mail_notification_with_history/data/mail_data.xml]: to be addressed

**Review by sebalix — APPROVED** [review]

**Pull request merged**